src/genie/libs/parser/iosxr/show_dampening.py

```diff
@@ -0,0 +1,241 @@
+""" show_dampening.py
+    supports commands:
+        * show im dampening
+        * show im dampening {interface}
+"""
+
+# Python
+import re
+# Metaparser
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Schema, Any, Optional
+
+# parser utils
+from genie.libs.parser.utils.common import Common
+
+# =============================================
+# Parser for 'show im dampening'
+# =============================================
+
+class ShowImDampeningSchema(MetaParser):
+    schema = {
+                'interface': {
+                    Any(): {
+                    'protocol': {
+                        Any(): {
+                        'capsulation': str,
+                        'penalty': int,
+                        'suppressed': str,
+                        }
+                    }
+                }
+            }
+        }
+                    
+class ShowImDampening(ShowImDampeningSchema):
+    """ Parser for show im dampening"""
+    
+    cli_command = 'show im dampening'
+
+    def cli(self,output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+            
+        result_dict = {}
+        result_dict.setdefault('interface', {})
+        
+        # GigabitEthernet0/0/0/1                                               2389 YES
+        # POS0/2/0/0                  <base>             ppp                      0 NO 
+        p1 = re.compile(r'^(?P<interface>.{1,26}) +(?P<prot>.{1,20}) +(?P<cap>.{1,10}) +(?P<pen>\d+) +(?P<sup>YES|NO)')
+
+        for line in out.splitlines():
+            if line:
+                line = line.strip()
+            else:
+                continue
+            
+            m = p1.match(line)
```

Comment:
Please add an example on the line above for what this regex matches. Makes it easier to see at a glance